test/serializers/attributes_test.rb

```diff
@@ -18,11 +18,31 @@ def test_attributes_definition
           @profile_serializer.class._attributes)
       end
 
+      def test_attributes
+        assert_equal({ name: 'Name 1', description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
       def test_attributes_with_fields_option
         assert_equal({ name: 'Name 1' },
           @profile_serializer.attributes(fields: [:name]))
       end
 
+      def test_attributes_with_fields_method
+        @profile_serializer.define_singleton_method(:include_attributes) do |included|
+          included - [:name]
+        end
+        assert_equal({ description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
+      def test_attributes_with_fields_option_and_method
+        @profile_serializer.define_singleton_method(:include_attributes) do |included|
+          included - [:name]
+        end
+        assert_equal({}, @profile_serializer.attributes(fields: [:name]))
```

Comment:
Also, as I wrote earlier, https://github.com/rails-api/active_model_serializers/pull/1141#issuecomment-142023735  fields is already available to the adapter and so this PR will **only** work for people getting attributes off the serializer directly which is not something this library, IMHO, supports.